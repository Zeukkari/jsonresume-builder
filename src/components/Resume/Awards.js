import React from 'react';

import {
  Box,
  DataTable,
  Text,
} from 'grommet';


const columns = [
  {
    property: "title",
    header: <Text>Award</Text>,
    primary: true,
  },
  {
    property: "date",
    header: "Date"
  },
  {
    property: "awarder",
    header: "Company"
  },
  {
    property: "summary",
    header: "Description"
  }
];

const Awards = ({ awards }) => {
  const renderItems = awards;

  return(
    <Box align="center" pad="large">
      <DataTable columns={columns} data={renderItems} />
    </Box>
  )
};

Awards.defaultProps = {
  language: []
}

export default Awards;