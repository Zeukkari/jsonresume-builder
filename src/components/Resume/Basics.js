import React from 'react';

import {
  Box,
  Heading,
  Paragraph,
} from 'grommet';

const Basics = ({ basics }) => {
  const name = basics.name || 'N/A';
  const label = basics.label || 'N/A';
  const summary = basics.summary || 'N/A';
  const email = basics.email || 'N/A';
  const website = basics.website || 'N/A';
  // const picture = basics.picture || 'N/A';
  // const profiles = basics.profiles || [];


  return (
    <Box>
      <Heading> Basics </Heading>
      <Paragraph>name: {name}</Paragraph>
      <Paragraph>label: {label}</Paragraph>
      <Paragraph>summary: {summary}</Paragraph>
      <Paragraph>email: {email} </Paragraph>
      <Paragraph>website: {website}</Paragraph>
    </Box>
  );
};

export default Basics;