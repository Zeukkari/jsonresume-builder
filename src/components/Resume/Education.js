import React from 'react';

import {
  Box,
  Heading,
} from 'grommet';


const Item = () => (
  <Box>
    <Heading>Education</Heading>
  </Box>
);


export default Item;