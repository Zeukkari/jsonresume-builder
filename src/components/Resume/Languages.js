import React from 'react'

import { Box, Heading, Text } from 'grommet'

const Languages = ({ languages }) => {
  const renderItems = languages.map(item => {
    return (
      <Box key="item.language">
        <Heading>{item.language}</Heading>
        <Heading>{item.fluency}</Heading>
      </Box>
    )
  })

  return (
    <Box margin="none" pad="none" border="all">
      <Box
        border={{
          side: 'all',
          color: 'red',
          size: 'xsmall',
          style: 'dashed',
        }}
        margin="none"
        pad="none"
      >
        <Heading
          level={3}
          size="medium"
          align="center"
          alignSelf="center"
          textAlign="center"
          margin="none"
          pad="none"
        >
          <Text>Languages</Text>
        </Heading>
        <Box>{renderItems}</Box>
      </Box>
    </Box>
  )
}

Languages.defaultProps = {
  language: [],
}

export default Languages
