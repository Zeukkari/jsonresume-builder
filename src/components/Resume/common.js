import React from 'react'

import { Box } from 'grommet'

export const ResumeSection = ({ children }) => (
  <Box
    border={{
      side: 'all',
      color: 'red',
      size: 'xsmall',
      style: 'dashed',
    }}
    margin="none"
    pad="none"
  >
    {children}
  </Box>
)
