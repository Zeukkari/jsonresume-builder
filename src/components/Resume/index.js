import Resume, { Loading } from './Resume'

export { Loading }
export default Resume
