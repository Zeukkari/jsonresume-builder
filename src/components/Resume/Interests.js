import React from 'react'

import { Grommet, DataTable, Heading, Text } from 'grommet'

import { ResumeSection, SubSection } from './common'

const columns = [
  {
    property: 'name',
    header: <Text>Name</Text>,
    primary: true,
  },
  {
    property: 'keywords',
    header: 'Keywords',
  },
]

const Interests = ({ interests }) => {
  const renderItems = interests.map(item => {
    const keywords = item.keywords.join()

    return { ...item, keywords }
  })

  return (
    <ResumeSection>
      <SubSection>
        <Heading> Interests </Heading>
      </SubSection>
      <Grommet>
        <SubSection>
          <DataTable columns={columns} data={renderItems} />
        </SubSection>
      </Grommet>
    </ResumeSection>
  )
}

Interests.defaultProps = {
  language: [],
}

export default Interests
