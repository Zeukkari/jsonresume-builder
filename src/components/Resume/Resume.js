import React from 'react'

import { Box, Heading } from 'grommet'

import Awards from './Awards'
import { Summary, Picture, Contact } from './Basics'
import Education from './Education'
import Interests from './Interests'
import Languages from './Languages'
import Publications from './Publications'
import References from './References'
import Skills from './Skills'
import Volunteer from './Volunteer'
import Work from './Work'

import {
  ResumeLayout,
  TopLayout,
  LeftColumnLayout,
  RightColumnLayout,
} from './Layout'

const Placeholder = ({
  work,
  basics,
  volunteer,
  education,
  awards,
  publications,
  skills,
  languages,
  interests,
  references,
  isValid,
}) => {
  if (!isValid) {
    return (
      <Box
        flex
        fill
        justify="center"
        wrap={true}
        height="medium"
        overflow="scroll"
        border={{ color: 'status-error', size: 'large' }}
      >
        <Heading>Error rendering</Heading>
      </Box>
    )
  }

  const {
    name,
    label,
    picture,
    email,
    phone,
    website,
    summary,
    location,
    profiles,
  } = basics

  return (
    <ResumeLayout>
      <TopLayout>
        <Box align="center" justify="center">
          <Picture picture={picture} />
        </Box>

        <Summary name={name} label={label} summary={summary} />

        <Contact
          email={email}
          phone={phone}
          website={website}
          location={location}
          profiles={profiles}
        />
      </TopLayout>

      <LeftColumnLayout>
        <Box>
          <Work work={work} />
        </Box>

        <Box>
          <Education education={education} />
        </Box>
      </LeftColumnLayout>

      <RightColumnLayout>
        <Box>
          <Skills skills={skills} />
        </Box>

        <Box>
          <Interests interests={interests} />
        </Box>

        <Box>
          <Awards awards={awards} />
        </Box>

        <Box>
          <Volunteer volunteer={volunteer} />
        </Box>

        <Box>
          <Publications publications={publications} />
        </Box>

        <Box>
          <Languages languages={languages} />
        </Box>

        <Box>
          <References references={references} />
        </Box>
      </RightColumnLayout>
    </ResumeLayout>
  )
}

export default Placeholder
