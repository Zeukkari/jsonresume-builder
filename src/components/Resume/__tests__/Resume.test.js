import React from 'react';
import renderer from 'react-test-renderer';

import Resume from '../Resume';

import defaultResume from '../../../util/defaultResume';


test('renders resume', () => {
  const data = defaultResume.projects;
  const component = renderer.create(
    <Resume data={{ isValid: true, value: data }} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});