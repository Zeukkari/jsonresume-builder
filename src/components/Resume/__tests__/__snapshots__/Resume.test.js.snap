// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders resume 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 bIxJtp"
>
  <h1
    className="StyledHeading-sc-1rdh4aw-0 jVpYfH"
  >
    Error rendering
  </h1>
</div>
`;
